homeassistant/components/ring/__init__.py

```diff
@@ -41,14 +39,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     def token_updater(token):
         """Handle from sync context when token is updated."""
-        run_callback_threadsafe(
-            hass.loop,
-            partial(
-                hass.config_entries.async_update_entry,
-                entry,
-                data={**entry.data, CONF_TOKEN: token},
-            ),
-        ).result()
+        hass.loop.call_soon_threadsafe(
+            hass.config_entries.async_update_entry,
+            entry,
+            data={**entry.data, CONF_TOKEN: token},
```

Comment:
I think we needed to keep the partial here since the `call_soon_threadsafe` doesn't support kwargs AFAICT

Reply:
Ah snap, you are right. It's even mentioned [in the docs](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.call_soon_threadsafe)

Reply:
Fix https://github.com/home-assistant/core/pull/107040

Reply:
Oh dang, I'm sorry, thanks for fixing it.

Reply:
no worries 👍